### PR TITLE
Using llvm cov for coverage

### PR DESCRIPTION
**Pull request by gnunicorn**
Let's try this out.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -132,18 +148,26 @@ jobs:
...
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+
       - name: Test
         uses: actions-rs/cargo@v1
         with:
-          command: nextest
-          args: run --workspace --exclude matrix-sdk-integration-testing
+          command: llvm-cov
+          args: nextest --lcov --output-path lcov.info run --workspace --exclude matrix-sdk-integration-testing
```
Comment (zecakeh) [inline .github/workflows/ci.yml:161]:
```suggestion
          args: --workspace --exclude matrix-sdk-integration-testing nextest --lcov --output-path lcov.info
```
`nextest` is the command so you don't need `run`.
There's also a weird bug (https://github.com/taiki-e/cargo-llvm-cov/issues/151) where the options that are forwarded to nextest need to be before the `nextest` command

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -254,5 +281,10 @@ jobs:
       - name: Test
         uses: actions-rs/cargo@v1
         with:
-          command: nextest
-          args: run -p matrix-sdk-integration-testing
+          command: llvm-cov
+          args: nextest --lcov --output-path lcov.info run -p matrix-sdk-integration-testing
```
Comment (zecakeh) [inline .github/workflows/ci.yml:289]:
```suggestion
          args: -p matrix-sdk-integration-testing nextest --lcov --output-path lcov.info
```

**Review by zecakeh — COMMENTED** [review]
I fixed the two first commands, but for xtask it's probably more complicated. take a look at https://github.com/taiki-e/cargo-llvm-cov/issues/151 for more details.

Basically without the fix it will always run coverage for the default crates with the default features.

**Comment by gnunicorn** [comment]
but generally codecov properly merges and updates the comment, which is a key I tried to test. next up: fixing what @zecakeh commented.

**Comment by gnunicorn** [comment]
While this showed we can generally do this, there's still a few things that need to be worked out (e.g. not including tests in the coverage) before we want to use this. Closing for now.

**Comment by jplatte** [comment]
In particular we would need https://github.com/rust-lang/rust/issues/84605.

**Pull request closed without merging**